File: ff4j-spring-boot-starter/src/main/java/org/ff4j/spring/boot/domain/PropertyStoreApiBean.java
/*
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Copyright 2013-2016 the original author or authors.
 */

package org.ff4j.spring.boot.domain;


import lombok.Getter;
import org.ff4j.cache.FF4jCacheProxy;
import org.ff4j.property.store.PropertyStore;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

/**
 * Created by Paul
 *
 * @author <a href="mailto:paul58914080@gmail.com">Paul Williams</a>
 */
public class PropertyStoreApiBean implements Serializable {

    private static final long serialVersionUID = 5459281574635411541L;

    @Getter
    private String type;

    @Getter
    private int numberOfProperties;

    @Getter
    private Set<String> properties = new HashSet<>();

    @Getter
    private CacheApiBean cache;

    public PropertyStoreApiBean(PropertyStore pStore) {
        type = pStore.getClass().getCanonicalName();
        if (pStore instanceof FF4jCacheProxy) {
            cache = new CacheApiBean(pStore);
        }
        properties = pStore.listPropertyNames();
        numberOfProperties = properties.size();
    }
}

File: ff4j-spring-services/src/main/kotlin/org/ff4j/services/domain/PropertyStoreApiBean.kt
/*-
 * #%L
 * ff4j-spring-services
 * %%
 * Copyright (C) 2013 - 2019 FF4J
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package org.ff4j.services.domain

import org.ff4j.cache.FF4jCacheProxy
import org.ff4j.property.store.PropertyStore
import java.io.Serializable
import java.util.*

/**
 * Created by Paul
 *
 * @author <a href="mailto:paul58914080@gmail.com">Paul Williams</a>
 */
class PropertyStoreApiBean : Serializable {

    companion object {
        private const val serialVersionUID = 5459281574635411541L
    }

    var type: String? = null
    var numberOfProperties: Int = 0
    var properties: Set<String> = HashSet()
    var cache: CacheApiBean? = null

    constructor() : super()

    constructor(propertyStore: PropertyStore) {
        this.type = propertyStore.javaClass.canonicalName
        if (isInstanceOfCache(propertyStore)) {
            this.cache = CacheApiBean(propertyStore)
        }
        this.properties = propertyStore.listPropertyNames()
        this.numberOfProperties = properties.size
    }

    private fun isInstanceOfCache(propertyStore: PropertyStore) = propertyStore is FF4jCacheProxy
}
